test/integration/container/multimaster_test.go

```diff
@@ -295,75 +325,90 @@ func TestMultimasterSetup(t *testing.T) {
 	}
 	yumRepoIP := sanitizeIP(run(t, "docker", "inspect", "yumrepo", "--format='{{.NetworkSettings.IPAddress}}'"))
 
-	// Start the footloose container "VMs" used for testing:
-	run(t, "footloose", "create", "-c", "../../../examples/footloose/centos7/docker/multimaster.yaml")
-	node0IP := sanitizeIP(run(t, "docker", "inspect", "centos-multimaster-node0", "--format='{{.NetworkSettings.IPAddress}}'"))
-	node1IP := sanitizeIP(run(t, "docker", "inspect", "centos-multimaster-node1", "--format='{{.NetworkSettings.IPAddress}}'"))
-	node2IP := sanitizeIP(run(t, "docker", "inspect", "centos-multimaster-node2", "--format='{{.NetworkSettings.IPAddress}}'"))
-	node3IP := sanitizeIP(run(t, "docker", "inspect", "centos-multimaster-node3", "--format='{{.NetworkSettings.IPAddress}}'"))
-
-	dirName := tempDir(t)
-	clusterYAML := saveToFile(t, dirName, "cluster.yaml", fmt.Sprintf(clusterYAML, registryIP, registryPort))
-	machinesYAML := saveToFile(t, dirName, "machines.yaml", fmt.Sprintf(machinesYAML, node0IP, node1IP, node2IP, node3IP))
-	_ = saveToFile(t, dirName, "repo-config.yaml", fmt.Sprintf(repoConfigMap, yumRepoIP))
-	_ = saveToFile(t, dirName, "docker-config.yaml", fmt.Sprintf(dockerConfigMap, registryIP, registryPort))
-
-	run(t, "../../../cmd/wksctl/wksctl", "apply",
-		fmt.Sprintf("--cluster=%s", clusterYAML), fmt.Sprintf("--machines=%s", machinesYAML),
-		fmt.Sprintf("--config-directory=%s", dirName),
-		"--verbose",
-		fmt.Sprintf("--controller-image=docker.io/weaveworks/wksctl-controller:%s", tag))
-
-	out := run(t, "../../../cmd/wksctl/wksctl", "kubeconfig",
-		fmt.Sprintf("--cluster=%s", clusterYAML), fmt.Sprintf("--machines=%s", machinesYAML))
-
-	var nodeList corev1.NodeList
-	for {
-		jsonOut := run(t, "kubectl", "get", "nodes", "-o", "json", fmt.Sprintf("--kubeconfig=%s", kubeconfig(out)))
-		if err := json.Unmarshal([]byte(jsonOut), &nodeList); err != nil {
-			log.Warnf("Error deserialising output of kubectl get nodes: %s", err)
-		}
-		log.Infof("The cluster currently has %d node(s)", len(nodeList.Items))
-		if len(nodeList.Items) == 4 {
-			break
-		}
-		time.Sleep(30 * time.Second)
+	tests := []struct {
```

Comment:
This looks like a refactoring of the test; please could you mention it in the commit description.

Reply:
Moved this to #221